lms/static/scripts/frontend_apps/components/dashboard/GradeIndicator.tsx

```diff
@@ -77,23 +85,40 @@ export default function GradeIndicator({
   const requiredCombined = config
     ? config.required_annotations + (config.required_replies ?? 0)
     : 0;
+  // Checking typeof lastGrade to avoid number zero to be treated as false
+  const hasLastGrade = typeof lastGrade === 'number';
+  const gradeHasChanged = lastGrade !== grade;
+  const showNewIndicator = !hasLastGrade || gradeHasChanged;
```

Comment:
In the case where there is no last grade, I expect that `lastGrade` is null/undefined and `grade` is a number, so `gradeHasChanged` will be true, meaning `!hasLastGrade` is not strictly required?

Reply:
Yep! You are right.